angrmanagement/plugins/log_human_activities/log_human_activities.py

```diff
@@ -0,0 +1,168 @@
+            project_md5=self.project_md5,
+            category=HumanActivityEnum.ClickInsn,
+            addr=qinsn.addr,
+            created_by=self.user,
+        )
+        self._log_list.append(insn_click)
+        l.info("Instruction %x is clicked", qinsn.addr)
+        return False
+
+    def handle_raise_view(self, view):
+        view_name = view.__class__.__name__
+        func = self._get_function_from_view(view)
+        if func is not None and not func.am_none:
+            func_name = func._name
+            addr = func.addr
+        else:
+            func_name = None
+            addr = None
+        raise_view = HumanActivity(
+            project=self.project_name,
+            project_md5=self.project_md5,
+            category=HumanActivityEnum.RaiseView,
+            view=view_name,
+            created_by=self.user,
+            function=func_name,
+            addr=addr
+        )
+        self._log_list.append(raise_view)
+        l.info("View %s is raised with function %s", view_name, func_name)
+
+    def handle_comment_changed(self, addr: int, cmt: str, new: bool, decomp: bool):
+        """
+        Log a user's activity of changing comment
+        @param new: T if a new comment. We don't log it in slacrs.
+        @param comp: T if comment is in decompiler view
+        """
+        comment_change = HumanActivity(
+            project=self.project_name,
+            project_md5=self.project_md5,
+            category=HumanActivityEnum.CommentChanged,
+            addr=addr,
+            cmt=cmt,
+            decomp=decomp,
+            created_by=self.user,
+        )
+        self._log_list.append(comment_change)
+        l.info("Comment is added at %x", addr)
+        return False
+
+    def handle_project_initialization(self):
+        """
+        Set project name
+        """
+        if self.workspace.instance.img_name is not None:
+            self.project_name = self.workspace.instance.img_name
+        else:
+            filename = self.workspace.instance.project.filename
+            self.project_name = filename
+            self.project_md5 = self.workspace.instance.project.loader.main_object.md5.hex()
+            l.info("Set project md5 to %s", self.project_md5)
+        l.info("Set project name to %s", self.project_name)
+
+    def _get_function_from_view(self, view):
+        if isinstance(view, Views.DisassemblyView):
+            return view._current_function
+        elif isinstance(view, Views.CodeView):
+            return view.function
+        elif isinstance(view, Views.ProximityView):
+            l.info("proximity view")
+            return view.function
+        else:
+            return None
+
+    def _commit_logs(self):
+        while True:
+            self.slacrs = Slacrs(database=Conf.checrs_backend_str)
+            self.session = self.slacrs.session()
+            while len(self._log_list) > 0:
+                l = self._log_list.pop()
+                self.session.commit(l)
```

Comment:
Can we add a bunch to the session and then commit at the end? That should speed things up.